container-images/ramalama/latest-cuda/Containerfile

```diff
@@ -0,0 +1,54 @@
+FROM nvidia/cuda:12.6.1-devel-ubi9
+
+# renovate: datasource=github-releases depName=huggingface/huggingface_hub extractVersion=^v(?<version>.*)
+ARG HUGGINGFACE_HUB_VERSION=0.25.0
+# renovate: datasource=github-releases depName=containers/omlmd extractVersion=^v(?<version>.*)
+ARG OMLMD_VERSION=0.1.4
+# renovate: datasource=git-refs depName=ggerganov/llama.cpp packageName=https://github.com/ggerganov/llama.cpp gitRef=master versioning=loose type=digest
+ARG LLAMA_CPP_SHA=32b2ec88bc44b086f3807c739daf28a1613abde1
+# renovate: datasource=git-refs depName=ggerganov/whisper.cpp packageName=https://github.com/ggerganov/whisper.cpp gitRef=master versioning=loose type=digest
+ARG WHISPER_CPP_SHA=5caa19240d55bfd6ee316d50fbad32c6e9c39528
+
+# vulkan-headers vulkan-loader-devel vulkan-tools glslc glslang python3-pip mesa-libOpenCL-$MESA_VER.aarch64
+RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    crb enable && \
+    dnf install -y epel-release && \
+    dnf --enablerepo=ubi-9-appstream-rpms install -y git procps-ng vim \
+      dnf-plugins-core python3-dnf-plugin-versionlock cmake gcc-c++ \
+      python3-pip && \
+    dnf clean all && \
+    rm -rf /var/cache/*dnf*
+
+RUN /usr/bin/python3 --version
+RUN pip install "huggingface_hub[cli]==${HUGGINGFACE_HUB_VERSION}"
+RUN pip install "omlmd==${OMLMD_VERSION}"
+
+# Build wouldnt complete couldnt find libcuda so made a systemlink
+# But this didnt work
+
+# RUN ln -s /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1
+# RUN LD_LIBRARY_PATH=/usr/local/cuda/lib64/stubs/
+
+# Disable cashing for debugging build
+
+# ENV GGML_CCACHE=0
+
+# Build wouldnt complete with cmake even with nvidia container toolkit installed
+
+RUN git clone https://github.com/ggerganov/llama.cpp && \
+    cd llama.cpp && \
+    git reset --hard ${LLAMA_CPP_SHA} && \
+    make -j $(nproc) GGML_CUDA=1 CUDA_ARCH=ALL && \
+    mv llama-cli /usr/bin/llama-cli && \
+    mv llama-server /usr/bin/llama-server && \
+    cd / && \
+    rm -rf llama.cpp
+
+RUN git clone https://github.com/ggerganov/whisper.cpp.git && \
+    cd whisper.cpp && \
+    git reset --hard ${WHISPER_CPP_SHA} && \
+    make -j $(nproc) && \
```

Comment:
Can we build whisper.cpp with GGML_CUDA=1 CUDA_ARCH=ALL also?

Reply:
Sure!